pkg/asset/agent/manifests/releaseimagearch.go

```diff
@@ -0,0 +1,156 @@
+)
+
+// Name returns a human friendly name for the asset.
+func (*ReleaseImageArch) Name() string {
+	return "Release Image Architecture"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ReleaseImageArch) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&agent.OptionalInstallConfig{},
+		&releaseimage.Image{},
+	}
+}
+
+// Generate release image info
+func (r *ReleaseImageArch) Generate(dependencies asset.Parents) error {
+	installConfig := &agent.OptionalInstallConfig{}
+	releaseImage := &releaseimage.Image{}
+	dependencies.Get(installConfig, releaseImage)
+
+	r.pullSecret = installConfig.Config.PullSecret
+	r.pullSpec = releaseImage.PullSpec
+	r.controlPlaneCPUArch = getarch.GoArch(string(installConfig.Config.ControlPlane.Architecture))
+	r.executer = &executer.CommonExecuter{}
+
+	return r.finish()
+}
+
+// Files returns the files generated by the asset.
+func (r *ReleaseImageArch) Files() []*asset.File {
+	// Return empty array because File will never be loaded.
+	return nil
+}
+
+// Load ReleaseImageArch from disk
+func (r *ReleaseImageArch) Load(f asset.FileFetcher) (bool, error) {
+	// The ReleaseImageArch will not be needed by another asset so load is noop.
+	// This is implemented because it is required by WritableAsset
+	return false, nil
+}
+
+func (r *ReleaseImageArch) finish() error {
+	err := r.validateReleaseImageArch()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func (r *ReleaseImageArch) validateReleaseImageArch() error {
+	// This requires the 'oc' command so make sure its available
+	_, err := exec.LookPath("oc")
+	var cmd string
+	if err != nil {
+		logrus.Error("Unable to validate release image architecture: \"oc\" command is not found")
+		return err
+	}
+
+	cmd = fmt.Sprintf(templateGetReleaseImageArch, r.pullSpec)
+	imageArch, err := execute(r.executer, r.pullSecret, cmd)
+	imageArch = strings.Trim(imageArch, "'")
+	if err == nil {
+		// Check the release image to see if it is multi, if so fail. We don't support multi release images at install time (yet).
+		if imageArch == "multi" {
+			err = fmt.Errorf("multi architecture release image %s is not supported", r.pullSpec)
+			return err
+		}
+		// if the release image isn't multi, then its single arch, and it must match the cpu architecture.
+		if imageArch == r.controlPlaneCPUArch {
+			logrus.Debugf("Supported architecturegit  %s found for the release image: %s", r.controlPlaneCPUArch, r.pullSpec)
+		} else {
+			err = fmt.Errorf("unsupported release image architecture. Expected ControlPlane Arch: %s Found Release Image Arch: %s ", r.controlPlaneCPUArch, imageArch)
+			return err
+		}
+	} else {
+		logrus.Error("Unable to validate the release image architecture")
+		return err
+	}
+
+	return nil
+}
+func execute(executer executer.Executer, pullSecret, command string) (string, error) {
```

Comment:
In general it would be recommend to avoid duplicating the code. https://github.com/openshift/installer/blob/e20821e3b213dbf6af19c2739dbab7c0c1e3c765/pkg/asset/agent/image/oc.go could be refactored to allow re-using it directly.

Reply:
agree.. this was the main reason for putting this PR to WIP.. trying to get a better sense of how this should look :)